In [1]:
import pandas as pd
import numpy as np
import os
import string

In [2]:
data_cleaned = pd.read_csv('/content/drive/MyDrive/data_cleaned_ty1.csv')

In [3]:
data_stemmed = pd.read_csv('/content/drive/MyDrive/data_stemmed_ty1.csv')

In [4]:
data_cleaned.head(10)

reviews  rating  recommendation
0  elbise görseldekinin birebir aynısı geldi her ...       5               1
1                              beden aldım cici bir        5               1
2  güzel bir elbise tam bir fiyat performans ürün...       5               1
3  kesinlikle güzel bir elbise beden alınız boy k...       5               1
4  güzel bir elbise kızım kuzeninin nişanında giy...       5               1
5  boy kiloyum minnoş elbise bayıldım çizmeyle ko...       5               1
6  mükemmel kalıp bayıldım i̇yiki yorumlara güven...       5               1
7  elbise harika bayıldım şık güzel boy kiloyum b...       5               1
8  çok asil zarif bir elbise belde taba kemerle s...       5               1
9  elbiseye bayıldım kiloyum yorumlara bakarak be...       5               1

In [5]:
data_stemmed.head(10)

reviews  rating  recommendation
0  elbise  görsel  birebir  aynı  gel  kadın  dol...       5               1
1                             beden  al  cici  bir         5               1
2  güzel  bir  elbise  tam  bir  fiyat  performan...       5               1
3  kesinlikle  güzel  bir  elbise  beden  al  boy...       5               1
4  güzel  bir  elbise  kız  kuzen  nişanın  gi  s...       5               1
5  boy  kilo  minnoş  elbise  bay  çizme  kombinl...       5               1
6  mükemmel  kalıp  bay  i̇yi  yorum  güven  bir ...       5               1
7  elbise  harika  bay  şık  güzel  boy  kilo  be...       5               1
8  çok  asil  zarif  bir  elbise  bel  taba  keme...       5               1
9  elbise  bay  kilo  yorum  bak  beden  al  i̇yi...       5               1

In [6]:
from sklearn.model_selection import train_test_split

In [7]:
y = data_cleaned['recommendation'].astype('category').cat.codes

X_train, X_test, y_train, y_test = train_test_split(list(data_cleaned['reviews']), y, 
                                                   test_size=0.2, 
                                                   random_state = 0, 
                                                   #stratify = y,
                                                   shuffle=True)

In [8]:
max_length = 500

In [9]:
from tensorflow.keras.layers.experimental.preprocessing import TextVectorization

vectorizer = TextVectorization(max_tokens=20000, output_sequence_length=max_length)
vectorizer.adapt(X_train)

In [10]:
vectorizer.get_vocabulary()[:6]

['', '[UNK]', 'beden', 'bir', 'güzel', 'elbise']

In [11]:
import os 

path_to_glove_file = os.path.join(
    os.path.expanduser("~"), "/content/drive/MyDrive/vocab.txt"
)

embeddings_index = {}
with open(path_to_glove_file) as f:
    for line in f:
        word, coefs = line.split(maxsplit=1)
        coefs = np.fromstring(coefs, "f", sep=" ")
        embeddings_index[word] = coefs

print("Found %s word vectors." % len(embeddings_index))

Found 570912 word vectors.


In [13]:
# Préparation de la matrix d'embedding

voc = vectorizer.get_vocabulary()
word_index = dict(zip(voc, range(len(voc))))

num_tokens = len(voc) + 2
embedding_dim = 300
hits = 0
misses = 0


embedding_matrix = np.zeros((num_tokens, embedding_dim))
for word, i in word_index.items():
    embedding_vector = embeddings_index.get(word)
    if embedding_vector is not None:
        embedding_matrix[i] = embedding_vector
        hits += 1
    else:
        misses += 1
print("Converted %d words (%d misses)" % (hits, misses))

Converted 2717 words (491 misses)


Burada kelimelerin kökeni bulunmadan önce toplam 2808 kelimenin 2717 tanesini yakaladığını söylüyor

In [14]:
y = data_stemmed['recommendation'].astype('category').cat.codes

X_train, X_test, y_train, y_test = train_test_split(list(data_stemmed['reviews']), y, 
                                                   test_size=0.2, 
                                                   random_state = 0, 
                                                   #stratify = y,
                                                   shuffle=True)

In [15]:
max_length = 500

In [16]:
from tensorflow.keras.layers.experimental.preprocessing import TextVectorization

vectorizer = TextVectorization(max_tokens=20000, output_sequence_length=max_length)
vectorizer.adapt(X_train)

In [17]:
vectorizer.get_vocabulary()[:6]

['', '[UNK]', 'beden', 'al', 'bir', 'güzel']

In [18]:
import os 

path_to_glove_file = os.path.join(
    os.path.expanduser("~"), "/content/drive/MyDrive/vocab.txt"
)

embeddings_index = {}
with open(path_to_glove_file) as f:
    for line in f:
        word, coefs = line.split(maxsplit=1)
        coefs = np.fromstring(coefs, "f", sep=" ")
        embeddings_index[word] = coefs

print("Found %s word vectors." % len(embeddings_index))

Found 570912 word vectors.


In [19]:
# Préparation de la matrix d'embedding

voc = vectorizer.get_vocabulary()
word_index = dict(zip(voc, range(len(voc))))

num_tokens = len(voc) + 2
embedding_dim = 300
hits = 0
misses = 0


embedding_matrix = np.zeros((num_tokens, embedding_dim))
for word, i in word_index.items():
    embedding_vector = embeddings_index.get(word)
    if embedding_vector is not None:
        embedding_matrix[i] = embedding_vector
        hits += 1
    else:
        misses += 1
print("Converted %d words (%d misses)" % (hits, misses))

Converted 1347 words (331 misses)


Burada ise kelime köklerini bulduktan sonra toplam kelime sayımız 1678 oldu ve bunlardan 1347 tanesini yakaladık. 